From c52c372019318e0426c75496c3903196d7a6cf27 Mon Sep 17 00:00:00 2001
From: Mika Laitio <lamikr@gmail.com>
Date: Fri, 30 Aug 2024 14:07:36 -0700
Subject: [PATCH 8/9] disable mpitest due to ubuntu 22.04 linking failures

- rocm-openmpi that is tried to link contains dependency
  to libpmix.so and on ubuntu 22.04 that is not linked
  automatically which causes unresolved symbols errors
- simply adding -lpmix did not help, so disabling the
  build of test now

Signed-off-by: Mika Laitio <lamikr@gmail.com>
---
 test/cpp/c10d/CMakeLists.txt | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/test/cpp/c10d/CMakeLists.txt b/test/cpp/c10d/CMakeLists.txt
index 5c8974836d..5f92023008 100644
--- a/test/cpp/c10d/CMakeLists.txt
+++ b/test/cpp/c10d/CMakeLists.txt
@@ -10,6 +10,7 @@ function(c10d_add_test test_src)
   add_executable(${test_name} "${test_src}")
   target_include_directories(${test_name} PRIVATE $<BUILD_INTERFACE:${TORCH_SRC_DIR}/csrc/distributed>)
   target_link_libraries(${test_name} ${ARGN})
+  message(STATUS "ARGN: " ${ARGN})
   if(NOT WIN32)
     target_link_libraries(${test_name} pthread)
   endif()
@@ -73,13 +74,15 @@ else()
   endif()
 endif()
 
-if(USE_MPI AND USE_C10D_MPI)
+if(USE_MPI_ROCM_SDK_DISABLED AND USE_C10D_MPI)
   add_definitions(-DMPIEXEC=${MPIEXEC})
   # MPI is a private dependency of libtorch, but the tests include some
   # private headers of libtorch, which in turn include MPI. As a hacky
   # alternative to making MPI a public dependency of libtorch, we make it
   # a private dependency of the tests as well.
   c10d_add_test(ProcessGroupMPITest.cpp torch_cpu MPI::MPI_CXX)
+  #get_filename_component(test_name ProcessGroupMPITest.cpp NAME_WE)
+  #target_link_libraries(${test_name} pmix)
   if(INSTALL_TEST)
     install(TARGETS ProcessGroupMPITest DESTINATION bin)
   endif()
-- 
2.43.0

